### ISSUE TITLE
why divide noise by batch size?

### ISSUE THREAD

**Issue by jason-jiankai** [pytorch/opacus#200]
params = (p for p in self.module.parameters() if p.requires_grad)
        for p, clip_value in zip(params, clip_values):
            noise = self._generate_noise(clip_value, p)
            if self.loss_reduction == "mean":
                noise /= batch_size
            p.grad += noise

the noise is added to the averaged gradient, why should it be divided by the batch size?
since the output is the final grad, should it be added with noise directly?

**Comment by ilyamironov** [comment]
Thank you for your interest! For consistency and discoverability, we prefer having all Q&A's on the dedicated forum: https://discuss.pytorch.org/c/opacus/29. You are welcome to ask your question there. I'm closing the issue for now.
